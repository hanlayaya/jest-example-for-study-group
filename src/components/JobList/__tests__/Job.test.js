import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import styles from '../Job.scss';

describe('<Job />', function () {
  let Job;

  beforeEach(() => {
    jest.resetModules();
    Job = require('../Job').default;

    this.params = {};

    this.makeSubject = () => {
      const { job } = this.params;

      return shallow(<Job job={job} />);
    };
  });

  describe('when it has complete data', () => {
    beforeEach(() => {
      this.params.job = {
        name: 'Frontend Developer',
        company: {
          brand: 'Partner Studio',
        },
        skill_tags: [{
          id: 1,
          name: 'JavaScript',
        }, {
          id: 2,
          name: 'CSS',
        }],
      };

      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(toJson(this.subject)).toMatchSnapshot();
    });

    it('should display `Partner Studio`', () => {
      expect(this.subject.find('h2').text()).toEqual('Partner Studio');
    });
  });

  describe('when it lack of tag data', () => {
    beforeEach(() => {
      this.params.job = {
        name: 'Frontend Developer',
        company: {
          brand: 'Partner Studio',
        },
        skill_tags: [],
      };

      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(toJson(this.subject)).toMatchSnapshot();
    });

    it('should not display any skill tag', () => {
      expect(this.subject.find(`.${styles['skill-tag']}`).length).toEqual(0);
    });
  });
});
