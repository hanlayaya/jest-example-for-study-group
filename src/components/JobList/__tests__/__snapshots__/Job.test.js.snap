// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Job /> when it has complete data should be same as snapshot 1`] = `
<div
  className="job"
>
  <h1>
    Frontend Developer
  </h1>
  <h2>
    Partner Studio
  </h2>
  <div
    className="skill-tags"
  >
    <span
      className="skill-tag"
    >
      JavaScript
    </span>
    <span
      className="skill-tag"
    >
      CSS
    </span>
  </div>
</div>
`;

exports[`<Job /> when it lack of tag data should be same as snapshot 1`] = `
<div
  className="job"
>
  <h1>
    Frontend Developer
  </h1>
  <h2>
    Partner Studio
  </h2>
  <div
    className="skill-tags"
  />
</div>
`;
